invenio/modules/news/views/webnews.py

```diff
@@ -0,0 +1,195 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015 CERN.
+#
+# Invenio is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""nwsToolTip Flask Blueprint."""
+
+from flask import(
+    Blueprint,
+    flash,
+    jsonify,
+    redirect,
+    render_template,
+    request,
+    session,
+)
+from flask.ext.menu import register_menu
+
+from invenio.ext.sqlalchemy import db
+
+from sqlalchemy.exc import IntegrityError
+
+from .. import config
+from ..encoder import Decode, Encode
+from ..models import NwsSTORY, NwsTAG, NwsToolTip
+
+blueprint = Blueprint(
+    'webnews',
+    __name__,
+    template_folder='../templates',
+    static_folder='../static'
+)
+# @register_menu(blueprint, 'main.webnews',config.CFG_WEBNEWS_ADMIN_MAIN_NAV)
+# @register_menu(blueprint, 'webnews',config.CFG_WEBNEWS_ADMIN_MAIN_NAV)
+
+
+@blueprint.route(config.CFG_WEBNEWS_MENU_INDEX)
+@register_menu(
+    blueprint,
+    'webnews.menu.search',
+    [
+        config.CFG_WEBNEWS_SEARCH_NAV_NAME,
+        'glyphicon glyphicon-search',
+        'general'
+    ]
+)
+def index():
+    """index function."""
+    try:
+        result = NwsSTORY.query.filter_by(
+            document_status='SHOW'
+        ).limit(5).all()
+        return render_template(
+            'search.html',
+            searchResult=result,
+            EncodeStr=Encode
+        )
+    except Exception:
+        db.create_all()
+        return redirect(config.CFG_WEBNEWS_MENU_INDEX)
+
+
+@blueprint.route(config.CFG_WEBNEWS_SEARCH, methods=['GET', 'POST'])
+def search():
+    """search function."""
+    if request.method == 'POST':
+        try:
+            result = NwsSTORY.query.filter(
+                NwsSTORY.title.contains(
+                    request.form['keywords']
+                ) | NwsSTORY.body.contains(request.form['keywords'])
+            ).filter_by(document_status='SHOW').all()
+            return render_template(
+                'search.html',
+                searchResult=result,
+                resultshow='block',
+                EncodeStr=Encode
+            )
+        except IntegrityError:
+            flash('Error')
```

Comment:
use correct flash message type and wrap the descriptive message into translatable string ...